src_features/generic_tx_parser/cmd_field.c

```diff
@@ -0,0 +1,50 @@
+#ifdef HAVE_GENERIC_TX_PARSER
+
+#include "cmd_field.h"
+#include "cx.h"
+#include "apdu_constants.h"
+#include "mem.h"
+#include "tlv.h"
+#include "tlv_apdu.h"
+#include "gtp_field.h"
+#include "cmd_tx_info.h"
+#include "gtp_tx_info.h"
+
+static bool handle_tlv_payload(const uint8_t *payload, uint16_t size, bool to_free) {
+    s_field field;
+    s_field_ctx ctx;
+
+    explicit_bzero(&field, sizeof(field));
+    explicit_bzero(&ctx, sizeof(ctx));
+    ctx.field = &field;
+    if (!tlv_parse(payload, size, (f_tlv_data_handler) &handle_field_struct, &ctx)) return false;
+    if (cx_hash_no_throw(get_fields_hash_ctx(), 0, payload, size, NULL, 0) != CX_OK) return false;
```

Comment:
We don't care here is the payload contains a Signature tag?

Reply:
This payload is not signed, since the signed TX info payload contains a hash of all the field structs. :+1: